core/tag.h

```diff
@@ -37,6 +37,27 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "platform.h"
 #include "limits.h"
 
+#ifdef GRAN_MICRO
```

Comment:
Are the flags `GRAN_MICRO` and `BIT_32` any different?  What are they meant to stand for? Perhaps more verbose flags would be better?

Reply:
The differences are subtle but necessary. GRAN_MICRO is used to determine what our fundamental unit of time is whereas BIT_32 tells us how large our data structure variable size can be. Arduino has both a fundamental unit of time of microseconds and only supports 32 bit length variable sizes (no long long support). I can clear up the flags a bit better.

Reply:
Though we could combine them together as a single flag, there may be use cases down the line where an embedded platform has millisecond fundamental unit of time or supports 32 bit variable size, so it may be fine to keep them split up.

Reply:
Changed GRAN_MICRO to MICROSECOND_TIME, kept BIT_32 the same